projects/plugins/jetpack/extensions/blocks/ai-image/edit.js

```diff
@@ -88,120 +93,127 @@ export default function Edit( { attributes, setAttributes, clientId } ) {
 		);
 	};
 
+	const ImageWithSelect = ( { image, withSelect = true } ) => {
+		return (
+			<Flex direction="column">
+				<FlexBlock>
+					<img
+						className="wp-block-ai-image-image"
+						src={ image }
+						alt=""
+						onClick={ () => setImageModal( image ) }
+					/>
+				</FlexBlock>
+				{ withSelect && (
+					<FlexBlock>
+						<Flex direction="column" style={ { 'align-items': 'center' } }>
+							<FlexItem>
+								<Button variant="primary" onClick={ () => saveImage( image ) }>
+									{ __( 'select', 'jetpack' ) }
+								</Button>
+							</FlexItem>
+						</Flex>
+					</FlexBlock>
+				) }
+			</Flex>
+		);
+	};
+
+	const saveImage = async image => {
+		if ( loadingImages ) {
+			return;
+		}
+		setLoadingImages( true );
+		setErrorMessage( null );
+
+		// First convert image to a proper blob file
+		const resp = await fetch( image );
+		const blob = await resp.blob();
+		const file = new File( [ blob ], 'jetpack_ai_image.png', {
+			type: 'image/png',
+		} );
+		// Actually upload the image
+		mediaUpload( {
+			filesList: [ file ],
+			onFileChange: ( [ img ] ) => {
+				if ( ! img.id ) {
+					// Without this image gets uploaded twice
+					return;
+				}
+				replaceBlock(
+					clientId,
+					createBlock( 'core/image', {
+						url: img.url,
+						caption: attributes.requestedPrompt,
+						alt: attributes.requestedPrompt,
+					} )
+				);
+			},
+			allowedTypes: [ 'image' ],
+			onError: message => {
+				// eslint-disable-next-line no-console
+				console.error( message );
+				setLoadingImages( false );
+			},
+		} );
+	};
+
 	return (
 		<div { ...useBlockProps() }>
-			{ ! loadingImages && errorMessage && (
-				<Placeholder
-					label={ __( 'AI Image', 'jetpack' ) }
-					notices={ [ <div>{ errorMessage }</div> ] }
-				>
-					<TextareaControl
-						label={ __( 'What would you like to see?', 'jetpack' ) }
-						value={ prompt }
-						placeholder={ placeholder }
-						onChange={ setPrompt }
-					/>
-					<Flex direction="row">
-						<FlexItem>
+			<Placeholder
+				label={ __( 'AI Image', 'jetpack' ) }
+				notices={ errorMessage && [ <div>{ errorMessage }</div> ] }
+			>
+				{ ! loadingImages && resultImages.length === 0 && (
+					<Flex expanded={ true }>
+						<FlexBlock>
+							<TextareaControl
+								label={ __( 'What would you like to see?', 'jetpack' ) }
+								placeholder={ placeholder }
+								allowedFormats={ [] }
+								onChange={ setPrompt }
+								rows={ 6 }
+							/>
 							<Button variant="primary" onClick={ submit }>
-								{ __( 'Retry', 'jetpack' ) }
+								{ errorMessage ? __( 'Retry', 'jetpack' ) : __( 'Submit', 'jetpack' ) }
```

Comment:
This is breaking the build I'm afraid:

```
msgid argument is not a string literal: __(E?"Retry":"Submit","jetpack")
```

The building tooling doesn't seem to like the translations in a ternary like that, so I would recommend moving them out of there, maybe like so (it's not as pretty, but it does seem to build):

```diff
diff --git a/projects/plugins/jetpack/extensions/blocks/ai-image/edit.js b/projects/plugins/jetpack/extensions/blocks/ai-image/edit.js
index 3ed9e51195..f970a7d0b2 100644
--- a/projects/plugins/jetpack/extensions/blocks/ai-image/edit.js
+++ b/projects/plugins/jetpack/extensions/blocks/ai-image/edit.js
@@ -71,6 +71,9 @@ export default function Edit( { attributes, setAttributes, clientId } ) {
 	const { replaceBlock } = useDispatch( blockEditorStore );
 	const [ errorMessage, setErrorMessage ] = useState( null );
 	const [ placeholder ] = useState( getRandomItem( dalleExamplePrompts ) );
+	const errorButtonText = __( 'Retry', 'jetpack' );
+	const successButtonText = __( 'Submit', 'jetpack' );
+	const submitButtonText = errorMessage ? errorButtonText : successButtonText;
 
 	const { mediaUpload } = useSelect( select => {
 		const { getSettings } = select( blockEditorStore );
@@ -175,7 +178,7 @@ export default function Edit( { attributes, setAttributes, clientId } ) {
 								rows={ 6 }
 							/>
 							<Button variant="primary" onClick={ submit }>
-								{ errorMessage ? __( 'Retry', 'jetpack' ) : __( 'Submit', 'jetpack' ) }
+								{ submitButtonText }
 							</Button>
 						</FlexBlock>
 					</Flex>

```

Reply:
Funny I didn't see any issue when building 🤔 

While you are here. Another question. My building keeps breaking until I do a `composer update` in `projects/plugins/jetpack`. Don't you push the new `composer.lock`?

Reply:
I see. You're talking about the GH actions build...

Reply:
See discussion in https://github.com/Automattic/jetpack/pull/28285

Reply:
> See discussion in #28285

Thanks! I didn't see those!
So it looks like an issue with the PHP version?

Reply:
> You're talking about the GH actions build...

Yes. We get no build errors when doing a dev build locally with `jetpack build plugins/jetpack`.

> So it looks like an issue with the PHP version?

Yup, looks like it. You would need to update to PHP 8.0. Although in the case of this PR, you shouldn't have to run any kind of `composer` commands really, since you're not making any changes to composer packages. You should be able to build and update the blocks in your PRs with `jetpack build plugins/jetpack` and `jetpack watch plugins/jetpack`. I don't believe you need any other build commands at the moment. Maybe `jetpack rsync` to test on WordPress.com simple or WoA.